nyxtools/flyer.py

```diff
@@ -0,0 +1,177 @@
+        self.vector.move()
+
+        return NullStatus()
+
+    def collect(self):
+        self.unstage()
+
+        now = ttime.time()
+        data = {
+            f"{self.detector.name}_image": self._datum_ids["data"],
+            "omega": self._datum_ids["omega"],
+        }
+        yield {
+            "data": data,
+            "timestamps": {key: now for key in data},
+            "time": now,
+            "filled": {key: False for key in data},
+        }
+
+    def collect_asset_docs(self):
+
+        asset_docs_cache = []
+
+        # Get the Resource which was produced when the detector was staged.
+        ((name, resource),) = self.detector.file.collect_asset_docs()
+
+        asset_docs_cache.append(("resource", resource))
+        self._datum_ids = DEFAULT_DATUM_DICT
+        # Generate Datum documents from scratch here, because the detector was
+        # triggered externally by the DeltaTau, never by ophyd.
+        resource_uid = resource["uid"]
+        # num_points = int(math.ceil(self.detector.cam.num_images.get() /
+        #                 self.detector.cam.fw_num_images_per_file.get()))
+
+        # We are currently generating only one datum document for all frames, that's why
+        #   we use the 0th index below.
+        #
+        # Uncomment & update the line below if more datum documents are needed:
+        # for i in range(num_points):
+
+        self._first_file = f"{resource['root']}/{resource['resource_path']}_00001.cbf"
+        if not os.path.isfile(self._first_file):
+            raise RuntimeError(f"File {self._first_file} does not exist")
+
+        # The pseudocode below is from Tom Caswell explaining the relationship between resource, datum, and events.
+        #
+        # resource = {
+        #     "resource_id": "RES",
+        #     "resource_kwargs": {},  # this goes to __init__
+        #     "spec": "AD-EIGER-MX",
+        #     ...: ...,
+        # }
+        # datum = {
+        #     "datum_id": "a",
+        #     "datum_kwargs": {"data_key": "data"},  # this goes to __call__
+        #     "resource": "RES",
+        #     ...: ...,
+        # }
+        # datum = {
+        #     "datum_id": "b",
+        #     "datum_kwargs": {"data_key": "omega"},
+        #     "resource": "RES",
+        #     ...: ...,
+        # }
+
+        # event = {...: ..., "data": {"detector_img": "a", "omega": "b"}}
+
+        for data_key in self._datum_ids.keys():
+            datum_id = f"{resource_uid}/{data_key}"
+            self._datum_ids[data_key] = datum_id
+            datum = {
+                "resource": resource_uid,
+                "datum_id": datum_id,
+                "datum_kwargs": {"data_key": data_key},
+            }
+            asset_docs_cache.append(("datum", datum))
+        return tuple(asset_docs_cache)
+
+    def _extract_metadata(self, field="omega"):
+        with fabio.open(self._first_file, "r") as cbf:
+            return cbf.pilatus_headers("omega")
```

Comment:
This method is not used anywhere, probably it's not needed.

Reply:
we agreed to keep this for now - we can delete it later (and won't affect anything since it isn't used)